src/api/mutation/use-wallet-migration.tsx

```diff
@@ -0,0 +1,11 @@
+import { useMutation } from '../../base';
+import { AugmentedMutationOptions } from '../../base/use-mutation';
+
+export const useWalletMigration = ({ ...props }: Omit<AugmentedMutationOptions<'wallet_migration'>, 'name'> = {}) => {
+    const { data, ...rest } = useMutation({ name: 'wallet_migration', ...props });
+
+    return {
+        data: data?.wallet_migration,
+        ...rest,
+    };
```

Comment:
Note: I didn't add the logic for refetching the data for `wallet_migration` because it should be handled on the consumer's side (since its already part of a 'business logic').
But one way to handle it would be (in consumer side):
```
    const { ... } = useWalletMigration({
        ...
            onSuccess: (data) => {
                 if (data?.wallet_migration?.state === 'in_progress') setTimeout(() => invalidate('wallet_migration'), 500)
    },
    })
```